packages/specs/pages/masp/masp-vp.mdx

```diff
@@ -0,0 +1,203 @@
+# MASP validity predicate (VP)
+
+The MASP can be an Namada established account with a validity predicate which handles the verification of shielded
+transactions.
+Similarly to zcash, the asset balance of the shielded pool itself is transparent - that is, from the transparent
+perspective, the MASP is just an account holding assets.
+
+The MASP VP has the following functions:
+
+- Accepts only valid transactions involving assets moving in or out of the pool.
+- Accepts valid shielded-to-shielded transactions, which don't move assets from the perspective of transparent Namada¹.
+- Publishes the note commitment and nullifier² reveal Merkle trees.
+
+To make this possible, the host environment¹ needs to provide verification primitives to VPs. One possibility is to
+provide a single high-level operation to verify transaction output descriptions² and proofs, but another is to provide
+cryptographic functions in the host environment and implement the verifier as part of the VP.
+1. what is the host environment?
+2. why are we using the word "descriptions" here? What is it? Which one are we of the two are we going to unpack?
```

Comment:
```suggestion

```

Reply:
I don't know what we want to replace this with. I have no valid definition for the 'host environment' @cwgoes ?
@CarloModicaPortfolio What do you mean by "Which one of the two are we going to unpack?" -- Which two?